src/commands/kv/key/put.rs

```diff
@@ -82,3 +72,78 @@ pub fn put(target: &Target, user: &GlobalUser, data: KVMetaData) -> Result<(), f
 
     Ok(())
 }
+
+fn get_response(
+    data: KVMetaData,
+    user: &GlobalUser,
+    url: &Url,
+) -> Result<reqwest::blocking::Response, failure::Error> {
+    let url_into_str = url.to_string();
+    let client = http::legacy_auth_client(user);
+    let res = match data.metadata {
+        Some(metadata) => {
+            let part = if data.is_file {
+                multipart::Part::file(&data.value)?
+            } else {
+                multipart::Part::text(data.value)
+            };
+            let form = multipart::Form::new()
+                .part("value", part)
+                .text("metadata", metadata.to_string());
+            client.put(&url_into_str).multipart(form).send()?
+        }
+        None => {
+            let value_body = get_request_body(data)?;
+            client.put(&url_into_str).body(value_body).send()?
+        }
+    };
+    Ok(res)
+}
+
+// If is_file is true, overwrite value to be the contents of the given
+// filename in the 'value' arg.
+fn get_request_body(data: KVMetaData) -> Result<Body, failure::Error> {
+    if data.is_file {
+        match &metadata(&data.value) {
+            Ok(file_type) if file_type.is_file() => {
+                let file = fs::File::open(&data.value)?;
+                Ok(file.into())
+            }
+            Ok(file_type) if file_type.is_dir() => failure::bail!(
+                "--path argument takes a file, {} is a directory",
+                data.value
+            ),
+            // last remaining value is symlink
+            Ok(_) => failure::bail!("--path argument takes a file, {} is a symlink", data.value),
```

Comment:
I'm a little confused as to why this prior code was calling out symlinks... I'd expect it to work, and I read this api doc https://doc.rust-lang.org/std/fs/struct.Metadata.html#method.is_file  as suggesting you should try opening the file rather than explicitly testing if it's a regular file.

Reply:
I just tried it on master, symlinks work. I can replace the error text to say
```
--path argument points to unknown entity
```

Reply:
done